src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/TciOnFdGrid.cpp

```diff
@@ -6,23 +6,36 @@
 #include <cstddef>
 
 #include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
 #include "DataStructures/Tensor/Tensor.hpp"
 #include "Evolution/DgSubcell/PerssonTci.hpp"
 #include "NumericalAlgorithms/Spectral/Mesh.hpp"
 
 namespace grmhd::ValenciaDivClean::subcell {
 bool TciOnFdGrid::apply(const Scalar<DataVector>& tilde_d,
                         const Scalar<DataVector>& tilde_tau,
+                        const tnsr::I<DataVector, 3, Frame::Inertial>& tilde_b,
                         const bool vars_needed_fixing, const Mesh<3>& dg_mesh,
                         const TciOptions& tci_options,
                         const double persson_exponent) noexcept {
-  return vars_needed_fixing or
-         min(get(tilde_d)) <
-             tci_options.minimum_rest_mass_density_times_lorentz_factor or
-         min(get(tilde_tau)) < tci_options.minimum_tilde_tau or
-         evolution::dg::subcell::persson_tci(tilde_d, dg_mesh, persson_exponent,
-                                             1.0e-18) or
-         evolution::dg::subcell::persson_tci(tilde_tau, dg_mesh,
-                                             persson_exponent, 1.0e-18);
+  bool cell_is_troubled =
+      vars_needed_fixing or
+      min(get(tilde_d)) <
+          tci_options.minimum_rest_mass_density_times_lorentz_factor or
+      min(get(tilde_tau)) < tci_options.minimum_tilde_tau or
+      evolution::dg::subcell::persson_tci(tilde_d, dg_mesh, persson_exponent,
+                                          1.0e-18) or
+      evolution::dg::subcell::persson_tci(tilde_tau, dg_mesh, persson_exponent,
+                                          1.0e-18);
+  if (tci_options.magnetic_field_cutoff.has_value() and not cell_is_troubled) {
+    const Scalar<DataVector> tilde_b_magnitude = magnitude(tilde_b);
+    cell_is_troubled =
+        cell_is_troubled or
```

Comment:
In this `if` you know `cell_is_troubled` is false so I would erase the redundant `cell_is_troubled or` from the RHS.